### PR TITLE
Add progress bar component

### PR DESCRIPTION
**Pull request by cavasinf**
## Description
Add progress bar component

progress bar

```twig
{{ progressBar() }}

{{ progressBar({current : 1}) }}

{{ progressBar({current : 2, max : 3, min : -1}) }}

{{ progressBar({current : 2, max : 3}) }}

{{ progressBar({current : 2, max : 3, color : 'bg-red'}, {backgroundColor : 'bg-grey'}) }}

{{ progressBar({current : 2, max : 3, color : 'bg-green'}, {progressSize : 'progress-sm'}) }}

{{ progressBar({current : 2, max : 3, color : 'bg-orange'}, {style : 'height: 20px;'}) }}

{{ progressBar({color : 'bg-red'}, {backgroundColor : 'bg-grey'}) }}

{{ progressBar({current :2, max: 5, color : 'bg-red', striped : true}) }}

{{ progressBar({current :2, max: 5, striped : true, animated: true}) }}

{{ progressBars([
    {current :15, max: 100},
    {current :30, max: 100, color: 'bg-success'},
    {current :20, max: 100, color: 'bg-danger'},
]) }}
```

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist
- [x] I updated the documentation (see [here](https://github.com/kevinpapst/TablerBundle/tree/master/docs))
- [x] I agree that this code will be published under the [MIT license](https://github.com/kevinpapst/TablerBundle/blob/master/LICENSE)

### CODE REVIEW

File: templates/components/progress-bar.html.twig [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    {% set _items = items ?? [] %}
+
+    <div class="progress {{ _backgroundColor }} {{ _progressSize }}" style="{{ _style }}">
+        {% for item in _items %}
+
+            {% set _value = item.current ?? 0 %}
+            {% set _max = item.max ?? 0 %}
+            {% set _min = item.min ?? 0 %}
+            {% set _indeterminate = item.indeterminate ?? _value == _max and _min == _max %}
+            {% set percent = _indeterminate or (_max - _min) == 0 ? 0 : ((_value - _min) * 100) / (_max - _min) %}
```
Comment (kevinpapst) [inline templates/components/progress-bar.html.twig:16]: not that I care deeply, but to be consequent shouldn't that have an underscore as well?

File: docs/components-progress-bar.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
+# Components
+
+This theme ships some components (as twig macros) that hide the complexity of rendering the same elements over and over again with the correct HTML.
+
+## ProgressBar (single)
+
+Progress Bar has been implemented to simplify the use of the Tabler Progress Bar component.
+`ProgressBar()` is a Shortcut of `ProgressBars()` with an array of ONE item.
```
Comment (kevinpapst) [inline docs/components-progress-bar.md:8]: Can we use `progress_bar` and `progress_bars` please.

We already have `status_dot` and `status_indicator`.

File: templates/components/progress-bar.html.twig (hunk 1 above)
Comment (cavasinf) [reply templates/components/progress-bar.html.twig:16]: `percent` is the only var that is not coming from user params. It is a calculated value.
So no `_` for him.

Do you want me to add it ?

File: docs/components-progress-bar.md (hunk 2 above)
Comment (cavasinf) [reply docs/components-progress-bar.md:8]: Sure 👍

File: templates/components/progress-bar.html.twig (hunk 1 above)
Comment (kevinpapst) [reply templates/components/progress-bar.html.twig:16]: Fine with me. But we should write all of our own rules down at some place.

**Review by kevinpapst — APPROVED** [review]
Looks good! 👍 
Just waiting with the merge, until I find time to create a demo page.

**Comment by kevinpapst** [comment]
I can't push changes 😢 time wasted...

File: docs/components-progress-bar.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+|    Parameter    | Description                                    |   Type   |    Default     |
+|:---------------:|------------------------------------------------|:--------:|:--------------:|
+|  progressSize   | Size of the progress bar                       | `string` | *empty string* |                                        
+| backgroundColor | background color of the progress bar container | `string` | *empty string* |                                        
+|      style      | Style to add to the progress bar container     | `string` | *empty string* | 
+
+### Usage
+
+```twig
+{% from '@Tabler/components/progress-bar.html.twig' import progress_bar %}
```
Comment (kevinpapst) [inline docs/components-progress-bar.md:43]:
```suggestion
{% from '@Tabler/components/progress_bar.html.twig' import progress_bar %}
```

File: docs/components-progress-bar.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+| Parameter | Description                   |   Type   | Default |
+|:---------:|:------------------------------|:--------:|:-------:|
+|   items   | Array of [Item](#Item) object | `array`  |  `[]`   |
+|  options  | [Options](#Option) object     | `object` |  `{}`   |
+
+### Usage
+
+```twig
+{% from '@Tabler/components/progress-bar.html.twig' import progress_bars %}
```
Comment (kevinpapst) [inline docs/components-progress-bar.md:81]:
```suggestion
{% from '@Tabler/components/progress_bar.html.twig' import progress_bars %}
```

**Review by kevinpapst — CHANGES_REQUESTED** [review]
Can you please rename the file and use underscore

**Comment by cavasinf** [comment]
It's not common for me to have underscores inside filename.
Looking at doc and layer files, they have hyphens, but components no.. bit disturbing.

**Comment by kevinpapst** [comment]
Yeah, true. Looks kind of consequent inconsequent 😁 
But at least all components and includes use hyphens, so filename matches the macro name.
Thanks for the change!

**Review by kevinpapst — APPROVED** [review]

**Pull request merged**